### PR TITLE
Add a server option to avoid loading pushed dates

### LINKED ISSUE #598: Policy bot stuck on `Commit hash does not have a pushed date`
Hello, we are seeing errors on policy bot where it's stuck on trying to get the `pushed date` 
image

Is there anything we can do to debug this issue?

Attached is the stack trace 

```go
failed to compute approval status: failed to list commits: Commit f3bed522f4 does not have a pushed date. This is usually caused by delays in the GitHub API
github.com/palantir/policy-bot/policy/approval.(*Rule).filteredCommits
	/home/runner/work/policy-bot/policy-bot/policy/approval/approve.go:345
github.com/palantir/policy-bot/policy/approval.(*Rule).IsApproved
	/home/runner/work/policy-bot/policy-bot/policy/approval/approve.go:204
github.com/palantir/policy-bot/policy/approval.(*Rule).Evaluate
	/home/runner/work/policy-bot/policy-bot/policy/approval/approve.go:141
github.com/palantir/policy-bot/policy/approval.(*RuleRequirement).Evaluate
	/home/runner/work/policy-bot/policy-bot/policy/approval/evaluate.go:63
github.com/palantir/policy-bot/policy/approval.(*OrRequirement).Evaluate
	/home/runner/work/policy-bot/policy-bot/policy/approval/evaluate.go:86
github.com/palantir/policy-bot/policy/approval.(*AndRequirement).Evaluate
	/home/runner/work/policy-bot/policy-bot/policy/approval/evaluate.go:146
github.com/palantir/policy-bot/policy/approval.(*evaluator).Evaluate
	/home/runner/work/policy-bot/policy-bot/policy/approval/evaluate.go:39
github.com/palantir/policy-bot/policy.evaluator.Evaluate
	/home/runner/work/policy-bot/policy-bot/policy/policy.go:80
github.com/palantir/policy-bot/server/handler.(*EvalContext).EvaluatePolicy
	/home/runner/work/policy-bot/policy-bot/server/handler/eval_context.go:125
github.com/palantir/policy-bot/server/handler.(*Details).ServeHTTP
	/home/runner/work/policy-bot/policy-bot/server/handler/details.go:132
github.com/bluekeyes/hatpear.Try.func1
	/home/runner/work/policy-bot/policy-bot/vendor/github.com/bluekeyes/hatpear/hatpear.go:84
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
goji%2eio.dispatch.ServeHTTP
	/home/runner/work/policy-bot/policy-bot/vendor/goji.io/dispatch.go:17
github.com/palantir/policy-bot/server/handler.RequireLogin.func1.1
	/home/runner/work/policy-bot/policy-bot/server/handler/login.go:95
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
goji%2eio.(*Mux).ServeHTTP
	/home/runner/work/policy-bot/policy-bot/vendor/goji.io/mux.go:74
goji%2eio.dispatch.ServeHTTP
	/home/runner/work/policy-bot/policy-bot/vendor/goji.io/dispatch.go:17
goji%2eio.(*Mux).ServeHTTP
	/home/runner/work/policy-bot/policy-bot/vendor/goji.io/mux.go:74
goji%2eio.dispatch.ServeHTTP
	/home/runner/work/policy-bot/policy-bot/vendor/goji.io/dispatch.go:17
github.com/bluekeyes/hatpear.Recover.func1.1
	/home/runner/work/policy-bot/policy-bot/vendor/github.com/bluekeyes/hatpear/hatpear.go:113
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
github.com/bluekeyes/hatpear.Catch.func1.1
	/home/runner/work/policy-bot/policy-bot/vendor/github.com/bluekeyes/hatpear/hatpear.go:60
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
github.com/palantir/go-baseapp/baseapp.AccessHandler.func1.1
	/home/runner/work/policy-bot/policy-bot/vendor/github.com/palantir/go-baseapp/baseapp/middleware.go:88
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
github.com/rs/zerolog/hlog.RequestIDHandler.func1.1
	/home/runner/work/policy-bot/policy-bot/vendor/github.com/rs/zerolog/hlog/hlog.go:187
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
github.com/palantir/go-baseapp/baseapp.NewMetricsHandler.func1.1
	/home/runner/work/policy-bot/policy-bot/vendor/github.com/palantir/go-baseapp/baseapp/middleware.go:55
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
github.com/rs/zerolog/hlog.NewHandler.func1.1
	/home/runner/work/policy-bot/policy-bot/vendor/github.com/rs/zerolog/hlog/hlog.go:29
net/http.HandlerFunc.ServeHTTP
	/opt/hostedtoolcache/go/1.20.4/x64/src/net/http/server.go:2122
goji%2eio.(*Mux).ServeHTTP
	/home/runner/work/policy-bot/policy-bot/vendor/goji.io/mux.go:74
```

### PR DESCRIPTION
**Pull request by asvoboda**
The functionality was broken by github 2023-07-01

Partially resolves #598

### CODE REVIEW

File: server/handler/eval_options.go [hunk 1]
```diff
@@ -45,6 +45,11 @@ type PullEvaluationOptions struct {
 	//
 	// TODO(bkeyes): remove in version 2.0
 	Deprecated_AppName string `yaml:"app_name"`
+
+	// As of 2023-07-01 the commit.pushedDate graphql field is removed from GitHub.
+	// !Warning!
+	// Setting this option effectively breaks all usage of the invalidate_on_push approval rule
+	DoNotLoadCommitPushedDate bool `yaml:"do_not_load_commit_pushed_date"`
```
Comment (agirlnamedsophia) [inline server/handler/eval_options.go:53]: Hi there! Thank you so much for this quick fix, my team and I really appreciate it. Will this option be available to set via an env var, similar to `shared_repository` (meaning something like `POLICYBOT_OPTIONS_DO_NOT_LOAD_COMMIT_PUSHED_DATE`)? We use env vars for our server configuration in our k8s deployment yaml.

Comment (DepickereSven) [reply server/handler/eval_options.go:53]: @agirlnamedsophia as far as I know to support this that would require a change in this piece of code to support it:
https://github.com/palantir/policy-bot/blob/fa1bb320a0c34797cb2b497c622208bd823b8eb4/server/handler/eval_options.go#L70-L75

Comment (DepickereSven) [reply server/handler/eval_options.go:53]: @agirlnamedsophia I created the following PR for it:
https://github.com/palantir/policy-bot/pull/600

File: README.md [hunk 2]
```diff
@@ -725,6 +725,21 @@ issue by using the `ignore_commits_by` option in combination with the
...
+`policy-bot` can be configured to invalidate approvals when a commit is pushed using
+the `invalidate_on_push` option. This option is disabled by default.
+
+However, in recent versions of GitHub after [2023-07-01 breaking changes][], policy-bot
+is unable to load pushedDate for commits. This means that it is unable to determine
+whether a commit was pushed after the approval was granted.
+
+A server option (`do_not_load_commit_pushed_date`) is provided to toggle loading the push
+date for commits. If enabled, policy-bot will not be able to determine whether a commit
+was pushed after an approval wast granted.
```
Comment (bluekeyes) [inline README.md]:
```suggestion
was pushed after an approval was granted.
```

**Review by bluekeyes — APPROVED** [review]
I think this is good as a quick fix. I'm worried at some point GitHub will start rejecting queries that request this field instead of just not returning a value for it, but I guess we can deal with that when it happens if we don't have a better fix by then.

**Comment by asvoboda** [comment]
Once this PR merges the feature will be available on our snapshot docker image for anyone interested in testing it out.

**Review by bluekeyes — APPROVED** [review]

**Comment by asvoboda** [comment]
👍

**Comment by z-x63** [comment]
Thanks! Would we need to remove all references of invalidate_on_push: true in .policy.yml as well?

Added below to the policy-bot config under options with the new snapshot image:
  do_not_load_commit_pushed_date: true
  
Am seeing this error: 
failed to filter candidates: failed to list commits: Commit 3e0d394f79 does not have a pushed date. This is usually caused by delays in the GitHub API

(moved to issue thread)

**Pull request merged**